From 2aaa6fc0e1d541ca4d06b3ee8ac7da3b4da312d6 Mon Sep 17 00:00:00 2001
From: Yannick Fertre <yannick.fertre@foss.st.com>
Date: Thu, 22 Feb 2024 16:00:52 +0100
Subject: [PATCH] dt-bindings: display: panel: Add Rocktech HX8394 LCD panel
 controller

Add bindings for the HX8394 LCD panel controller.
(Used on 5.5" 720x1280 TFT LCD panel).

Change-Id: I192631e30e0b784b72a47e60c3d813e7c0f0dc66
Signed-off-by: Yannick Fertre <yannick.fertre@foss.st.com>
---
 .../display/panel/rocktech,hx8394.yaml        | 56 +++++++++++++++++++
 1 file changed, 56 insertions(+)
 create mode 100644 Documentation/devicetree/bindings/display/panel/rocktech,hx8394.yaml

--- /dev/null
+++ b/Documentation/devicetree/bindings/display/panel/rocktech,hx8394.yaml
@@ -0,0 +1,56 @@
+# SPDX-License-Identifier: (GPL-2.0-only OR BSD-2-Clause)
+%YAML 1.2
+---
+$id: http://devicetree.org/schemas/display/panel/rocktech,hx8394.yaml#
+$schema: http://devicetree.org/meta-schemas/core.yaml#
+
+title: Rocktech Semiconductor Corporation HX8394 5.5" 720p MIPI-DSI TFT LCD panel
+
+maintainers:
+  - YANNICK FERTRE <yannick.fertre@foss.st.com>
+
+description: |
+  The Rocktech Semiconductor Corporation HX8394 is a 5.5" 720x1280 TFT LCD
+  panel connected using a MIPI-DSI video interface.
+
+allOf:
+  - $ref: panel-common.yaml#
+
+properties:
+
+  compatible:
+    const: rocktech,hx8394
+
+  reg:
+    maxItems: 1
+    description: DSI virtual channel
+
+  backlight: true
+  enable-gpios: true
+  port: true
+  power-supply: true
+
+  reset-gpios:
+    maxItems: 1
+
+additionalProperties: false
+
+required:
+  - compatible
+  - power-supply
+  - reg
+
+examples:
+  - |
+    dsi {
+      #address-cells = <1>;
+      #size-cells = <0>;
+      panel@0 {
+        compatible = "rocktech,hx8394";
+        reg = <0>;
+        reset-gpios = <&gpiof 15 0>;
+        power-supply = <&v1v8>;
+        backlight = <&pwm_backlight>;
+      };
+    };
+...
